Infinite loop in `ssort()`

```cpp
In file included from SAWriter.cpp:8:0:
/home/UNIXHOME/cdunn/repo/gh/blasr/libcpp/alignment/suffixarray/ssort.hpp: In function ‘int ssort(SAIndex*, SAIndex*)’:
/home/UNIXHOME/cdunn/repo/gh/blasr/libcpp/alignment/suffixarray/ssort.hpp:303:23: warning: comparison of unsigned expression >= 0 is always true [-Wtype-limits]
         for(i=n; --k>=0; ) {  /* (2) order */
                       ^
```
The loop can never terminate, so this is a bug. This code needs to be fixed or deleted.

Note that this function is global and lacks a namespace, so anything could call it. The bug might have been introduced as long ago as 1997 at Lucent.

My suggested solution: `abort()` at line 302, and test **SAWriter**. Then remove lines 303-333.

--- a/alignment/suffixarray/ssort.hpp
+++ b/alignment/suffixarray/ssort.hpp
@@ -299,7 +299,9 @@ ssort(SAIndex a[], SAIndex s[])
                 if(!(pl[i]&ORIG))
                     finish(2);
         }
-
+        abort(); // else infinite loop! probably never called. ~cdunn 2016
+#ifdef DEAD_CODE_20160111
+        // Compiler would warn here (because of the infinite loop):
         for(i=n; --k>=0; ) {		/* (2) order */
             j = pl[k];
             do
@@ -331,6 +333,7 @@ ssort(SAIndex a[], SAIndex s[])
         }
         if(++k >= n)
             break;
+#endif  // DEAD_CODE
     }
 
     for(i=0; i<n; i++)

